### PR TITLE
new parser

### PR DESCRIPTION
**Pull request by smitsialis**
## Description
Added a new parser for "show cable tdr interface {interface}" command
This was originally part of PR#593 which became too old.  Therefore closed it and opened this PR.

## Motivation and Context
Parser did not previously exist

## Impact (If any)
NIL

## Screenshots:
image

## Checklist:

- [x] I have updated the changelog.
- [ ] I have updated the documentation (If applicable).
- [x] I have added tests to cover my changes (If applicable).
- [x] All new and existing tests passed.
- [x] All new code passed compilation.

### CODE REVIEW

File: src/genie/libs/parser/iosxe/show_cable_tdr_interface.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
+'''show_cable_tdr_interface.py
```
Comment (danielgraziano) [inline src/genie/libs/parser/iosxe/show_cable_tdr_interface.py:1]: please move this into a file called `show_cable.py`

File: src/genie/libs/parser/iosxe/show_cable_tdr_interface.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            ('interface'): str,
+            ('speed'): str,
+            ('date'): str,
+            ('time'): str,
+            'pairs': {
+                Any() : {
+                    ('length'): int,
+                    ('tolerance'): int,
+                    ('remote_pair'): str,
+                    ('status'): str,
```
Comment (danielgraziano) [inline src/genie/libs/parser/iosxe/show_cable_tdr_interface.py:33]: please remove brackets around all these keys in the schema:
`('interface'): str,` -> `'interface': str,`

File: src/genie/libs/parser/iosxe/show_cable_tdr_interface.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+class ShowCableTdrInterface(ShowCableTdrIntSchema):
+    """
+    Parser for:
+            show cable tdr interface {interface}
+    """
+
+    cli_command = 'show cable tdr interface {interface}'
+
+    def cli(self, interface="", output=None):
```
Comment (danielgraziano) [inline src/genie/libs/parser/iosxe/show_cable_tdr_interface.py:47]:
```suggestion
    def cli(self, interface=None, output=None):
```

File: src/genie/libs/parser/iosxe/show_cable_tdr_interface.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        tdr_dict = {}
+
+        # TDR test last run on: January 09 16:19:29
+        #
+        # Interface Speed Local pair Pair length        Remote pair Pair status
+        # --------- ----- ---------- ------------------ ----------- --------------------
+        # Gi1/0/1   auto  Pair A     0    +/- 1  meters N/A         Open
+        #                 Pair B     1    +/- 1  meters N/A         Open
+        #                 Pair C     0    +/- 1  meters N/A         Open
+        #                 Pair D     1    +/- 1  meters N/A         Open
```
Comment (danielgraziano) [inline src/genie/libs/parser/iosxe/show_cable_tdr_interface.py:61]: Please remove

**Review by danielgraziano — CHANGES_REQUESTED** [review]
Looks good besides those few comments :)

File: src/genie/libs/parser/iosxe/show_cable_tdr_interface.py (hunk 1 above)
Comment (smitsialis) [reply src/genie/libs/parser/iosxe/show_cable_tdr_interface.py:1]: @danielgraziano  Stupid question.  I'm I just renaming the file from show_cable_tdr_interface.py to show_cable.py (moving the code)?  Or is it just moving that one line to show_cable.py

Comment (GerriorL) [reply src/genie/libs/parser/iosxe/show_cable_tdr_interface.py:1]: You'd want to rename the whole file. Calling it show_cable.py leaves it open to having other `show cable` commands added to it in the future.

**Comment by smitsialis** [comment]
All changes were made weeks ago.  Can this MR be approved?

**Review by domachad — APPROVED** [review]

**Review by GerriorL — APPROVED** [review]

**Pull request merged**